src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -1627,19 +1648,11 @@ public override BoundNode VisitCall(BoundCall node)
                 (ParameterSymbol parameter, _) = GetCorrespondingParameter(i, parameters, argsToParamsOpt, expanded);
                 AttributeAnnotations annotations = parameter?.FlowAnalysisAnnotations ?? AttributeAnnotations.None;
 
-                // We'll ignore NotNullWhenFalse that is misused in metadata
-                if ((annotations & AttributeAnnotations.NotNullWhenFalse) != 0 &&
-                    parameter.ContainingSymbol.GetTypeOrReturnType().SpecialType != SpecialType.System_Boolean)
-                {
-                    annotations &= ~AttributeAnnotations.NotNullWhenFalse;
-                }
+                // Ignore NotNullWhenTrue that is misused
+                annotations = removeMisusedNotNullWhenSense(parameter, annotations, sense: true);
```

Comment:
>removeMisusedNotNullWhenSense [](start = 30, length = 29)

I would expect these to both be removed at the same time.

Reply:
I didn't understand this comment.

---
In reply to: [191578194](https://github.com/dotnet/roslyn/pull/26828#discussion_r191578194) [](ancestors = 191578194)

Reply:
I would expect the implementation of removeMisusedNotNullWhenSense to remove both kinds of annotations in one shot rather than dealing with them separately.

---
In reply to: [191825413](https://github.com/dotnet/roslyn/pull/26828#discussion_r191825413) [](ancestors = 191825413,191578194)